### PR TITLE
chore(ui): add ARIA functionality into Podman Machine resources

### PR DESCRIPTION
**Pull request by odockal**
### What does this PR do?
It adds ARIA functionality in Resources Connection side in podman desktop.
### What issues does this PR fix or reference?
#3934

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte [hunk 1]
```diff
@@ -153,9 +153,9 @@ function getLoggerHandler(
 
 {#if connectionStatus}
   {#if connection.lifecycleMethods && connection.lifecycleMethods.length > 0}
-    <div class="mt-2 relative">
+    <div class="mt-2 relative" role="group" aria-label="Connection Actions">
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte]: it's not clear why we need to move some group definition at an upper level

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte [hunk 2]
```diff
@@ -441,17 +441,17 @@ function hasAnyConfiguration(provider: ProviderInfo) {
...
           <PreferencesConnectionsEmptyRendering
             message="{provider.emptyConnectionMarkdownDescription}"
             hidden="{provider.containerConnections.length > 0 || provider.kubernetesConnections.length > 0}" />
           {#each provider.containerConnections as container}
             {@const peerProperties = new PeerProperties()}
-            <div class="px-5 py-2 w-[240px]">
+            <div class="px-5 py-2 w-[240px]" role="region" aria-label="{container.name}">
               <div class="float-right">
-                <Tooltip tip="{provider.name} details" bottom>
+                <Tooltip tip="{provider.name} Details" bottom>
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: AFAIK this change is not related to ARIA

**Review by benoitf — COMMENTED** [review]
some changes in the label name (like details --> Details or the opposite, etc) are failing unit tests that rely on these current definition

Title might be confusing as it says 'add ARIA functionality' but it
- move some existing definition
- rename some labels

to me, adding ARIA should only include new stuff else it's a refactoring, a renaming etc

File: packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte (hunk 1 above)
Comment (odockal) [reply packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte]: So they would be on the same level.

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte (hunk 2 above)
Comment (odockal) [reply packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: that is right. will fix.

File: packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte (hunk 1 above)
Comment (odockal) [reply packages/renderer/src/lib/preferences/PreferencesConnectionActions.svelte]: From testing point of view it does not matter after all. will revert.

**Comment by odockal** [comment]
> to me, adding ARIA should only include new stuff else it's a refactoring, a renaming etc

For me it seems that former labels, especially aria-labels were not really intended to provide complex ARIA functionality, more just like randomly placed into UI into some parts, so I was thinking that I am actually adding it. How about `ARIA functionality expansion` or `Enhancement and refactoring of ARIA functionality` would that fits better what I am doing her and helps to identify kind of changes?

File: packages/renderer/src/lib/ui/ConnectionStatus.svelte [hunk 3]
```diff
@@ -58,5 +58,5 @@ $: statusStyle = statusesStyle.get(status) || {
 };
 </script>
 
-<div aria-label="Connection Status Icon" class="{roundIconStyle} {statusStyle.bgColor}"></div>
-<span aria-label="Connection Status Label" class="{labelStyle} {statusStyle.txtColor}">{statusStyle.label}</span>
+<div aria-label="connection-status-icon" class="{roundIconStyle} {statusStyle.bgColor}"></div>
+<span aria-label="connection-status-label" class="{labelStyle} {statusStyle.txtColor}">{statusStyle.label}</span>
```
Comment (benoitf) [inline packages/renderer/src/lib/ui/ConnectionStatus.svelte:62]: to me labels are human labels not with `-` no ?

Comment (odockal) [reply packages/renderer/src/lib/ui/ConnectionStatus.svelte:62]: This is the state before, I am reverting it in order to do not mix adding - refactoring unless we agree on how to handle that. Otherwise, I would definitely go and fix such label as I encounter them when doing changes. Addressing them at once could be hard. So if it is ok to make also these changes even though they are on different place than is PR about, I would change it then.

Comment (odockal) [reply packages/renderer/src/lib/ui/ConnectionStatus.svelte:62]: This specific might require standalone pr, since also unit tests require updates...

**Review by benoitf — APPROVED** [review]
